hnn_core/lfp.py

```diff
@@ -0,0 +1,221 @@
+        H = a.dot(b) / dis
+
+        # total longitudinal distance from start of compartment
+        L = H + dis
+
+        # if a.dot(b) < 0, projection will fall on the
+        # compartment.
+        if H < 0:
+            H = -H
+
+        # distance ^ 2 of electrode to end
+        r_sq = np.linalg.norm(ele_pos - end) ** 2
+
+        # phi
+        num = np.sqrt(H ** 2 + r_sq) - H
+        denom = np.sqrt(H ** 2 + r_sq) - L
+        phi = 1. / dis * np.log(num / denom)
+
+    # [dis]: um; [sigma]: S / m
+    # [phi / sigma] = [1/dis] / [sigma] = 1 / [dis] x [sigma]
+    # [dis] x [sigma] = um x (S / m) = 1e-6 S
+    # transmembrane current returned by _ref_i_membrane_ is in [nA]
+    # ==> 1e-9 A x (1 / 1e-6 S) = 1e-3 V = mV
+    # ===> multiply by 1e3 to get uV
+    return 1000.0 * phi / (4.0 * np.pi * sigma)
+
+
+class _LFPElectrode:
+    """LFP electrode class.
+
+    Parameters
+    ----------
+    coord : tuple
+        The (x, y, z) coordinates (in um) of the LFP electrode.
+    sigma : float
+        Extracellular conductivity, in S/m, of the assumed infinite,
+        homogeneous volume conductor that the cell and electrode are in.
+    pc : instance of h.ParallelContext()
+        ParallelContext instance for running in parallel
+    cvode : instanse of h.CVode
+        Multi order variable time step integration method.
+    method : str
+        'psa' (default), i.e., point source approximation or line source
+        approximation, i.e., 'lsa'
+
+    Attributes
+    ----------
+    lfp_t : instance of h.Vector
+        The LFP time instances.
+    lfp_v : instance of h.Vector
+        The LFP voltage.
+    imem_vec : instance of h.Vector
+        The transmembrane ionic current.
+
+    Notes
+    -----
+    See Table 5 in http://jn.physiology.org/content/104/6/3388.long for
+    measured values of sigma in rat cortex (note units there are mS / cm)
+    """
+
+    def __init__(self, coord, sigma=0.3, pc=None, cvode=None, method='psa'):
+
+        secs_in_network = _get_all_sections()  # ordered list of h.Sections
+        # np.array of number of segments for each section, ordered as above
+        self.segment_counts = _get_segment_counts(secs_in_network)
+
+        # pointers assigned to _ref_i_membrane_ at each EACH segment below
+        self.imem_ptrvec = h.PtrVector(self.segment_counts.sum())
+        # placeholder into which pointer values are read on each sim step
+        self.imem_vec = h.Vector(int(self.imem_ptrvec.size()))
+        # transfer resistances, same length as membrane current pointers
+        self.r_transfer = np.empty((int(self.imem_ptrvec.size()), ),
+                                   dtype=np.int)
+        self.pc = pc
+        if self.pc is None:
+            self.pc = h.ParallelContext()
+
+        # Enables fast calculation of transmembrane current (nA) at each
+        # segment
+        self.bscallback = cvode.extra_scatter_gather(0, self.callback)
+
+        xfer_resistance_list = list()
```

Comment:
```suggestion
        xfer_resistance = list()
```

adding object type to variable name is usually a little redundant because you can figure out pretty easily from the operations (e.g., append, extend etc) you do to the object what kind of object it is.